src/HotChocolate/Language/src/Language.SyntaxTree/ObjectValueNode.cs

```diff
@@ -68,27 +66,7 @@ public bool Equals(ObjectValueNode? other)
             return true;
         }
 
-        if (other.Fields.Count == Fields.Count)
-        {
-            IEnumerator<ObjectFieldNode> otherFields = other.Fields
```

Comment:
This was also bad ... I know its not from you. The thing here is that if we have two object ordered differently they are not the same.